widgetsnbextension/src/manager.js

```diff
@@ -63,24 +75,29 @@ var WidgetManager = function (comm_manager, notebook) {
     this.comm_manager = comm_manager;
 
     var widget_md = notebook.metadata.widgets
-    if (widget_md && widget_md['application/vnd.jupyter.widget-state+json']) {
-        this.set_state(notebook.metadata.widgets['application/vnd.jupyter.widget-state+json'])
-    }
 
-    // Register with the comm manager.
+    // Steps that needs to be done:
+    // 1. Register comm target
+    // 2. Get any widget state from the kernel and open comms with existing state
+    // 3. Check saved state for widgets, and restore any that would not overwrite
+    //    any live widgets.
+
+    // Register with the comm manager. (1)
     this.comm_manager.register_target(this.comm_target_name, _.bind(this.handle_comm_open,this));
 
-    // Attempt to reconstruct any live comms by requesting them from the back-end.
+    // Attempt to reconstruct any live comms by requesting them from the back-end (2).
     var that = this;
-    var backed_widgets_loaded = this._get_comm_info().then(function(comm_ids) {
+    var backed_widgets_loaded = Promise.resolve().then(function() {
```

Comment:
I don't think we need to create an extra promise. `_get_comm_info` returns a promise, so that should work fine, right?